### PR TITLE
feat: Upgrade Java version to 17 and spring boot upgrade

### PR DESCRIPTION
**Pull request by Athy17**
409-Stacks: Upgrade Java modules

#### 📲 What

A description of the change.

#### 🤔 Why

Why it's needed, background context.

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

**Review by ekartsonakisamido — APPROVED** [review]

File: java/src/main/resources/application-azure.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 azure:
   application-insights:
     instrumentation-key: xxxxxx
-    enabled: true
+    enabled: false
```
Comment (ekartsonakisamido) [inline java/src/main/resources/application-azure.yml:4]: Don't we need this?

Comment (Athy17) [reply java/src/main/resources/application-azure.yml:4]: sorry yes. Will revert this

File: api-tests/pom.xml [hunk 2]
```diff
@@ -141,8 +142,8 @@
         </dependency>
         <dependency>
             <groupId>org.hamcrest</groupId>
-            <artifactId>hamcrest-all</artifactId>
-            <version>${hamcrest.all.version}</version>
+            <artifactId>hamcrest</artifactId>
+            <version>${hamcrest.version}</version>
```
Comment (JuliaNezEnsono) [inline api-tests/pom.xml:146]: Should be ${hamcrest.all.version} as defined on line 37. Or line 37 should be renamed

**Review by ElvenSpellmaker — APPROVED** [review]

**Pull request merged**